core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -1473,6 +1476,11 @@ private static void configureHonourErrorMappingsWhenPolicyAppliedOnOperation() {
                                                 minMuleVersion("4.5.0"));
   }
 
+  private static void registerFeatureOnMinMuleVersion(Feature feature) {
+    FeatureFlaggingRegistry featureFlaggingRegistry = FeatureFlaggingRegistry.getInstance();
+    featureFlaggingRegistry.registerFeatureFlag(feature, minMuleVersion(feature.getEnabledByDefaultSince()));
+  }
```

Comment:
I think we should eventually refactor this class to use this function instead of all the many individual feature registration functions that are so repetitive.
Additionally, there is no reason to hard-code the `minMuleVersion` during registration because we actually have it already defined in the `Feature` itself as its `enabledByDefaultSince` property. This could lead to mismatch issues.
But I don't want to mix all those changes in this same PR.